File: payment-service/src/main/java/com/rainbowforest/paymentservice/service/PaymentService.java
package com.rainbowforest.paymentservice.service;

import com.rainbowforest.paymentservice.dto.PaymentRequest;
import com.rainbowforest.paymentservice.dto.PaymentResponse;

import java.util.Map;

public interface PaymentService {
    PaymentResponse createPayment(PaymentRequest request);
    PaymentResponse getPaymentStatus(Long paymentId);
    PaymentResponse getPaymentStatusByTxnRef(String txnRef);
    void handleVNPayIPN(Map<String, String> params);
    PaymentResponse handleVNPayReturn(Map<String, String> params);
    PaymentResponse verifyVNPayReturn(Map<String, String> params);
    void refundPayment(Long paymentId);
    void refundPaymentByOrderId(Long orderId);
}

File: frontend/src/services/paymentService.js
import axiosClient from "../api/axiosClient";

export const createPayment = (payload) => {
  return axiosClient.post("/api/payments/vnpay/create", payload);
};

export const getPaymentStatus = (paymentId) => {
  return axiosClient.get(`/api/payments/${paymentId}`);
};

export const getPaymentStatusByTxnRef = (txnRef) => {
  return axiosClient.get(`/api/payments/vnpay/status/${txnRef}`);
};

export const verifyVNPayReturn = (payload) => {
  return axiosClient.post("/api/payments/vnpay/verify-return", payload);
};
